### PR TITLE
Put for Farm

### PR DESCRIPTION
**Pull request by JPautz**

### CODE REVIEW

File: webServer/src/test/java/com/jnicho/quickpick/FarmTests.java [hunk 1]
```diff
@@ -181,6 +190,53 @@ public void testBadAddManagerNotAManager() throws Exception {
...
+        List<Device> devices = new ArrayList<>();
+        users.add(user);
+        devices.add(device);
+
+        farm.setManagers(users);
+        farm.setDevices(devices);
+        farm = farmRepository.save(farm);
+
+        //assertFalse(farm.getManagers().isEmpty());
+        //assertFalse(farm.getDevices().isEmpty());
```
Comment (mdavis60) [inline webServer/src/test/java/com/jnicho/quickpick/FarmTests.java]: remove comments bro

File: webServer/src/main/java/com/jnicho/quickpick/controllers/FarmController.java [hunk 2]
```diff
@@ -64,6 +69,68 @@
         return ResponseEntity.ok(null);
     }
 
+    @Transactional
+    @RequestMapping(value = "/farm/{id}", method = RequestMethod.PUT)
+    public ResponseEntity<String> putFarm(@PathVariable("id") Long id, @RequestBody Farm farm) {
+
+        if(!farm.isValidFarm())
+            return ResponseEntity.status(400).body("Farm needs a name and address");
```
Comment (johnnicholson) [inline webServer/src/main/java/com/jnicho/quickpick/controllers/FarmController.java]: do you have to send a fully valid farm to update one field (that's fine, just making sure that was your intention)

File: webServer/src/main/java/com/jnicho/quickpick/controllers/FarmController.java [hunk 3]
```diff
@@ -64,6 +69,68 @@
...
+        farmToEdit.setName(farm.getName());
+
+        farmToEdit.setAllowClassifying(farm.isAllowClassifying());
+        farmToEdit.setAllowIdentifying(farm.isAllowIdentifying());
+        farmToEdit.setAllowPicking(farm.isAllowPicking());
+
+        farmToEdit.setMinClassificationsToFinalize(farm.getMinClassificationsToFinalize());
+        farmToEdit.setMinIdentificationsToFinalize(farm.getMinIdentificationsToFinalize());
+        farmToEdit.setMinSimilarPointsToFinalize(farm.getMinSimilarPointsToFinalize());
+
```
Comment (johnnicholson) [inline webServer/src/main/java/com/jnicho/quickpick/controllers/FarmController.java]: Any range check on all of the numbers above

File: webServer/src/main/java/com/jnicho/quickpick/controllers/FarmController.java [hunk 4]
```diff
@@ -64,6 +69,68 @@
...
+            else {
+                m.setFarm(farmToEdit);
+                managers.add(m);
+            }
+        }
+
+        farmToEdit.getDevices().clear();
+        List<Device> devices = new ArrayList<>();
+        for (Device d: farm.getDevices()) {
+            d.setFarmName(farmToEdit.getName());
```
Comment (johnnicholson) [inline webServer/src/main/java/com/jnicho/quickpick/controllers/FarmController.java]: wait, wait, wait, wait... this is still a string. ahhhhhhhhhh

File: webServer/src/main/java/com/jnicho/quickpick/controllers/FarmController.java (hunk 2 above)
Comment (JPautz) [reply webServer/src/main/java/com/jnicho/quickpick/controllers/FarmController.java]: yes

File: webServer/src/main/java/com/jnicho/quickpick/controllers/FarmController.java (hunk 4 above)
Comment (JPautz) [reply webServer/src/main/java/com/jnicho/quickpick/controllers/FarmController.java]: yeah :(

File: webServer/src/main/java/com/jnicho/quickpick/controllers/FarmController.java (hunk 3 above)
Comment (JPautz) [reply webServer/src/main/java/com/jnicho/quickpick/controllers/FarmController.java]: currently no, will be fixed with upcoming PR next sprint with error handling refactor

**Pull request merged**